File: Beecrowd_Parte5/Times_2370.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Definição da estrutura Aluno
typedef struct {
    char nome[15]; // Assumindo que o nome tem no máximo 14 caracteres (incluindo '\0')
    int habilidade;
} Aluno;

// Função de comparação para qsort (ordenação por habilidade decrescente)
int comparar(const void *a, const void *b) {
    Aluno *alunoA = (Aluno *)a;
    Aluno *alunoB = (Aluno *)b;
    return alunoB->habilidade - alunoA->habilidade;
}

int main() {
    int N, T;
    scanf("%d %d", &N, &T);
    getchar(); // Consumir o newline após os inteiros

    // Array de alunos
    Aluno alunos[N];

    // Leitura dos alunos
    for (int i = 0; i < N; i++) {
        scanf("%s %d", alunos[i].nome, &alunos[i].habilidade);
    }

    // Ordenar os alunos por habilidade (decrescente)
    qsort(alunos, N, sizeof(Aluno), comparar);

    // Arrays de times
    char times[T][N][15]; // times[T][N][15] para armazenar até N alunos com nomes de até 14 caracteres
    int numAlunosPorTime[T]; // Contador de alunos por time

    // Inicializar o contador de alunos por time
    for (int i = 0; i < T; i++) {
        numAlunosPorTime[i] = 0;
    }

    // Distribuir os alunos nos times
    for (int i = 0; i < N; i++) {
        int timeIndex = i % T; // Selecionar o time alternadamente
        strcpy(times[timeIndex][numAlunosPorTime[timeIndex]], alunos[i].nome);
        numAlunosPorTime[timeIndex]++;
    }

    // Ordenar os nomes dos alunos nos times
    for (int i = 0; i < T; i++) {
        qsort(times[i], numAlunosPorTime[i], sizeof(times[i][0]), (const void *)strcmp);
    }

    // Imprimir os times
    for (int i = 0; i < T; i++) {
        printf("Time %d\n", i + 1);
        for (int j = 0; j < numAlunosPorTime[i]; j++) {
            printf("%s\n", times[i][j]);
        }
        printf("\n"); // linha em branco após cada time
    }

    return 0;
}

File: Beecrowd_Parte5/Times_2370.java
import java.util.*;

public class Main {

    static class Aluno {
        String nome;
        int habilidade;

        public Aluno(String nome, int habilidade) {
            this.nome = nome;
            this.habilidade = habilidade;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int N = scanner.nextInt();
        int T = scanner.nextInt();
        scanner.nextLine(); // Consumir o newline após os inteiros

        List<Aluno> alunos = new ArrayList<>();

        // Leitura dos alunos
        for (int i = 0; i < N; i++) {
            String nome = scanner.next();
            int habilidade = scanner.nextInt();
            alunos.add(new Aluno(nome, habilidade));
        }

        // Ordenar os alunos pela habilidade (decrescente)
        Collections.sort(alunos, (a1, a2) -> Integer.compare(a2.habilidade, a1.habilidade));

        // Array de times
        List<List<String>> times = new ArrayList<>();
        for (int i = 0; i < T; i++) {
            times.add(new ArrayList<>());
        }

        // Distribuir os alunos nos times
        for (int i = 0; i < N; i++) {
            Aluno aluno = alunos.get(i);
            int timeIndex = i % T; // Seleciona o time alternadamente
            times.get(timeIndex).add(aluno.nome);
        }

        // Ordenar os nomes dos alunos nos times
        for (List<String> time : times) {
            Collections.sort(time);
        }

        // Imprimir os times
        for (int i = 0; i < T; i++) {
            System.out.println("Time " + (i + 1));
            for (String nome : times.get(i)) {
                System.out.println(nome);
            }
            System.out.println(); // linha em branco após cada time
        }

        scanner.close();
    }
}
